src/snowflake/snowpark/modin/plugin/compiler/snowflake_query_compiler.py

```diff
@@ -13430,3 +13430,88 @@ def dt_strftime(self, date_format: str) -> None:
         ErrorMessage.not_implemented(
             "Snowpark pandas doesn't yet support the method 'Series.dt.strftime'"
         )
+
+    def pct_change(
+        self,
+        periods: int = 1,
+        fill_method: Literal["backfill", "bfill", "pad", "ffill", None] = "pad",
+        limit: Optional[int] = None,
+        freq: Optional[Union[pd.DateOffset, timedelta, str]] = None,
+        axis: Axis = 0,
+        **kwargs: Any,
+    ) -> "SnowflakeQueryCompiler":
+        """
+        Fractional change between the current and a prior element.
+
+        Computes the fractional change from the immediately previous row by default.
+        This is useful in comparing the fraction of change in a time series of elements.
+
+        Parameters
+        ----------
+        periods : int, default 1
+            Periods to shift for forming percent change.
+
+        fill_method : {'backfill', 'bfill', 'pad', 'ffill'}, default 'pad'
+            How to handle NAs before computing percent changes.
+
+            Deprecated since version 2.1: All options of fill_method are deprecated except fill_method=None.
+
+        limit : int, optional
+            The number of consecutive NAs to fill before stopping.
+
+            Snowpark pandas does not support this parameter.
+
+            Deprecated since version 2.1.
+
+        freq : DateOffset, timedelta, or str, optional
+            Increment to use from time series API (e.g. ‘ME’ or BDay()).
+
+            Snowpark pandas does not currently support this parameter.
+
+        axis : Axis, default 0
+            This is not part of the documented `pct_change` API, but pandas forwards kwargs like this
+            to `shift`. To avoid unnecessary JOIN operations, we cannot compositionally use `QueryCompiler.shift`,
+            and instead have to validate the axis argument here.
+
+            Snowpark pandas currently only supports axis=0.
+        """
+        # `periods` is validated by the frontend
+        if limit is not None:
+            ErrorMessage.not_implemented(
+                "Snowpark pandas DataFrame/Series.pct_change does not support the 'limit' parameter since it is deprecated in pandas"
+            )
+        if freq is not None:
+            ErrorMessage.not_implemented(
+                "Snowpark pandas DataFrame/Series.pct_change does not yet support the 'freq' parameter"
+            )
+        if axis != 0:
+            ErrorMessage.not_implemented(
+                "Snowpark pandas DataFrame/Series.pct_change does not yet support axis=1"
+            )
+        frame = self._modin_frame
+        if fill_method is not None:
+            frame = self.fillna(
+                self_is_series=False, method=fill_method, axis=0
```

Comment:
How do we know `self_is_series=False`?

Reply:
`self_is_series` is only used when the fill value argumen to `fillna` is a Series or DF, in which case the operation is mapped to a where + notna().